renderer.ts

```diff
@@ -173,16 +173,33 @@ async function postProcessRenderedHTML(plugin: PandocPlugin, inputFile: string,
     for (let span of Array.from(wrapper.querySelectorAll('span.internal-embed'))) {
         let src = span.getAttribute('src');
         if (src) {
-            const subfolder = inputFile.substring(adapter.getBasePath().length);  // TODO: this is messy
-            const file = plugin.app.metadataCache.getFirstLinkpathDest(src, subfolder);
+            const link_split = parseLinktext(src);
+            const subfolder = inputFile.substring(adapter.getBasePath().length); // TODO: this is messy
+            const file = plugin.app.metadataCache.getFirstLinkpathDest(link_split.path, subfolder);
+
             try {
                 if (parentFiles.indexOf(file.path) !== -1) {
                     // We've got an infinite recursion on our hands
                     // We should replace the embed with a wikilink
                     // Then our link processing happens afterwards
                     span.outerHTML = `<a href="${file}">${span.innerHTML}</a>`;
                 } else {
-                    const markdown = await adapter.read(file.path);
+                    let markdown = await adapter.read(file.path);
```

Comment:
I'd have to look into the specifics, but it is generally recommended to use the vault instead of the adapter API because it is highly optimised. That's not an issue here, but could be a change for later.

Reply:
I guess I could add that patch, it's small enough. Merging this for now.